src/nvim/lua/executor.c

```diff
@@ -1406,6 +1406,7 @@ char_u *nlua_register_table_as_callable(typval_T *const arg)
   lua_State *const lstate = nlua_enter();
 
   int top = lua_gettop(lstate);
+  (void)top;
```

Comment:
What about this?

```suggestion
#ifndef NDEBUG
  int top = lua_gettop(lstate);
#endif
```

Reply:
Ahhhh, I see now. Very cool. Thanks, that's what I was looking for!